compiler/src/main/java/com/squareup/anvil/compiler/codegen/AnnotationReferenceExtensions.kt

```diff
@@ -0,0 +1,37 @@
+package com.squareup.anvil.compiler.codegen
+
+import com.squareup.anvil.compiler.internal.argumentType
+import com.squareup.anvil.compiler.internal.classDescriptor
+import com.squareup.anvil.compiler.internal.findAnnotationArgument
+import com.squareup.anvil.compiler.internal.getAnnotationValue
+import com.squareup.anvil.compiler.internal.reference.AnnotationReference
+import com.squareup.anvil.compiler.internal.reference.AnnotationReference.Descriptor
+import com.squareup.anvil.compiler.internal.reference.AnnotationReference.Psi
+import com.squareup.anvil.compiler.internal.reference.AnvilCompilationExceptionAnnotationReference
+import com.squareup.anvil.compiler.internal.reference.ClassReference
+import com.squareup.anvil.compiler.internal.reference.toClassReference
+import com.squareup.anvil.compiler.internal.requireFqName
+import org.jetbrains.kotlin.psi.KtClassLiteralExpression
+import org.jetbrains.kotlin.resolve.constants.KClassValue
+
+internal fun AnnotationReference.parentScope(): ClassReference {
+  return when (this) {
```

Comment:
What are your thoughts on defining this method in `AnnotationReference` so we can avoid the `when`?

Reply:
At the moment the utilities are defined in the compiler-utils module, but they're only used in the compiler module. I don't really want to expose this function in the compiler-utils modules and decided therefore to move it into the compiler module. In the compiler module we can only use extension functions.